airflow/contrib/hooks/gcs_hook.py

```diff
@@ -405,23 +328,13 @@ def get_size(self, bucket, object):
         self.log.info('Checking the file size of object: %s in bucket: %s',
                       object,
                       bucket)
-        service = self.get_conn()
-        try:
-            response = service.objects().get(
-                bucket=bucket,
-                object=object
-            ).execute()
-
-            if 'name' in response and response['name'][-1] != '/':
-                # Remove Directories & Just check size of files
-                size = response['size']
-                self.log.info('The file size of %s is %s bytes.', object, size)
-                return size
-            else:
-                raise ValueError('Object is not a file')
-        except HttpError as ex:
-            if ex.resp['status'] == '404':
-                raise ValueError('Object Not Found')
+        client = self.get_conn()
+        bucket = client.get_bucket(bucket)
+        blob = bucket.blob(blob_name=object)
```

Comment:
This code looks like the `get_blob` method code.
Is this duplication intentional?
https://github.com/googleapis/google-cloud-python/blob/master/storage/google/cloud/storage/bucket.py#L691-L706

But in this case in not effective. It's do 2 calls to external API. 

I write a sample script:
```python
client = storage.Client()

bucket = client.get_bucket("instance-mb-test-1")
blob = bucket.get_blob('file-1.bin')
print("Blob size: ", blob.size)
```
On the screen a have a message:
```DEBUG:urllib3.util.retry:Converted retries value: 3 -> Retry(total=3, connect=None, read=None, redirect=None, status=None)
DEBUG:google.auth.transport.requests:Making request: POST https://oauth2.googleapis.com/token
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): oauth2.googleapis.com:443
DEBUG:urllib3.connectionpool:https://oauth2.googleapis.com:443 "POST /token HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.googleapis.com:443
DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "GET /storage/v1/b/instance-mb-test-1?projection=noAcl HTTP/1.1" 200 447
DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "GET /storage/v1/b/instance-mb-test-1/o/file-1.bin HTTP/1.1" 200 753
Blob size:  104960000
```
It's confirm that you implementation do a two API calls + 1 for authorization. 
I am proposing that you use the code:
```python
client = storage.Client()
bucket = storage.Bucket(client, "instance-mb-test-1")
blob = bucket.get_blob('file-1.bin')
print("Blob size: ", blob.size)
```
It's do one API call:
```python
DEBUG:urllib3.util.retry:Converted retries value: 3 -> Retry(total=3, connect=None, read=None, redirect=None, status=None)
DEBUG:google.auth.transport.requests:Making request: POST https://oauth2.googleapis.com/token
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): oauth2.googleapis.com:443
DEBUG:urllib3.connectionpool:https://oauth2.googleapis.com:443 "POST /token HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.googleapis.com:443
DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "GET /storage/v1/b/instance-mb-test-1/o/file-1.bin HTTP/1.1" 200 753
Blob size:  104960000
```
It is important to optimize this methos, because it is often used in a loop, and therefore the number of queries is significant.

Reply:
This is inspired from Google's code and examples:

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/storage/cloud-client/snippets.py

I don't get why there are 2 calls for one and not the other, may be I am missing something. Because it can either first get bucket object and then create a blob or create a bucket object and then get blob, looks same to me.

Reply:
I have changed it in few places, though

Reply:
The `get_bucket` method executes `reload` method, so it does additional API request.
1. First API call: 
https://github.com/googleapis/google-cloud-python/blob/master/storage/google/cloud/storage/client.py#L227
2. Second API call:
https://github.com/googleapis/google-cloud-python/blob/master/storage/google/cloud/storage/bucket.py#L702

Implementation of `reload` is available: https://github.com/googleapis/google-cloud-python/blob/master/storage/google/cloud/storage/_helpers.py#L110-L132

Reply:
There is an additional advantage in using both `get_bucket` and `get_blob`. The `get_bucket` method raises an error if the Bucket does not exists. The `get_blob` just returns None if the object doesn't exist.

I like this:
```
client = storage.Client()

bucket = client.get_bucket("instance-mb-test-1")
blob = bucket.get_blob('file-1.bin')
print("Blob size: ", blob.size)
```

Let me know what you think.

Reply:
**Case (1)**:

![image](https://user-images.githubusercontent.com/8811558/55713039-c8a3b480-59e7-11e9-8041-806906af1a29.png)

**Case (2)**:
![image](https://user-images.githubusercontent.com/8811558/55713075-d822fd80-59e7-11e9-832a-b0e9bb6637d8.png)


I would like to sticket with `get_bucket` so that we get a meaningful error, rather than just **None**

Reply:
Also, `get_blob()` doesn't contain `blob.reload` in the latest stable `google-cloud-storage` release  (**1.14.0**)

https://github.com/googleapis/google-cloud-python/blob/storage-1.14.0/storage/google/cloud/storage/bucket.py#L642

The link you pasted is for master branch and has not yet made through in release :)